pkg/util/helper/work_test.go

```diff
@@ -17,13 +17,272 @@ limitations under the License.
 package helper
 
 import (
-	"reflect"
+	"context"
 	"testing"
+	"time"
 
+	"github.com/stretchr/testify/assert"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
 )
 
+func TestCreateOrUpdateWork(t *testing.T) {
+	scheme := runtime.NewScheme()
+	assert.NoError(t, workv1alpha1.Install(scheme))
+	assert.NoError(t, workv1alpha2.Install(scheme))
+
+	tests := []struct {
+		name         string
+		existingWork *workv1alpha1.Work
+		workMeta     metav1.ObjectMeta
+		resource     *unstructured.Unstructured
+		wantErr      bool
+		verify       func(*testing.T, client.Client)
+	}{
+		{
```

Comment:
https://github.com/karmada-io/karmada/blob/5140eb1e1c757c782d54a60a57b19947e0391848/pkg/util/helper/work.go#L43-L53

This part seems not be covered by any test cases

Reply:
Ohh they were executed but not tested. Thanks!